common/beerocks/bwl/dummy/ap_wlan_hal_dummy.cpp

```diff
@@ -128,7 +136,20 @@ bool ap_wlan_hal_dummy::disable() { return true; }
 
 bool ap_wlan_hal_dummy::set_start_disabled(bool enable, int vap_id) { return true; }
 
-bool ap_wlan_hal_dummy::set_channel(int chan, int bw, int center_channel) { return true; }
+bool ap_wlan_hal_dummy::set_channel(int chan, int bw, int center_channel)
+{
+    m_radio_info.channel = chan;
+    m_radio_info.bandwidth =
+        beerocks::utils::convert_bandwidth_to_int((beerocks::eWiFiBandwidth)bw);
+    m_radio_info.vht_center_freq = center_channel;
+    m_radio_info.is_dfs_channel  = son::wireless_utils::is_dfs_channel(chan);
+    std::stringstream value;
+    value << "channel: " << chan << std::endl;
```

Comment:
nitpick, because I don't know what `flush()` means in `std::stringstream` ([example reference](https://youtu.be/GMqQOEZYVJQ))
```suggestion
    value << "channel: " << chan << '\n';    
```

Reply:
Sorry, I can't be bother to go through 11 minutes of chatter to get a simple point. Could you summarize the argument he is trying to make?

(note: we're currently using endl everywhere so it's quite a big thing you're proposing)

Reply:
yes, sure:
`std::endl` does more than just adding a `new-line` to the stream. It actually asks the stream to `flush` the already collected data into the underlying media.
So when it comes to disk it takes _much_ longer to `std::endl` instead of `\n`.
What I mentioned that I have no idea what does this extra work actually do when it comes to `std::stringstream` - it might end up with a simple nop (no operation).

Reply:
Indeed for strstream it's a nop.

And the flush behaviour is usually what you want. We, at least, use it only in the context of logging (and in beerocks_cli for user interaction). Line-buffered is exactly what you want in those cases.